src/Cardano/Address/Style/Jormungandr.hs

```diff
@@ -0,0 +1,314 @@
+
+instance HasNetworkDiscriminant Jormungandr where
+    type NetworkDiscriminant Jormungandr = (AddressDiscrimination, NetworkTag)
+    addressDiscrimination = fst
+    networkTag = snd
+
+instance PaymentAddress Jormungandr where
+    paymentAddress discrimination k = unsafeMkAddress $
+        invariantSize expectedLength $ BL.toStrict $ runPut $ do
+            putWord8 firstByte
+            putByteString (getPublicKey $ getKey k)
+      where
+          firstByte = case addressDiscrimination @Jormungandr discrimination of
+              RequiresNetworkTag -> 0x83
+              RequiresNoTag -> 0x03
+          expectedLength = 1 + publicKeySize
+
+instance DelegationAddress Jormungandr where
+    delegationAddress discrimination paymentKey stakingKey = unsafeMkAddress $
+        invariantSize expectedLength $ BL.toStrict $ runPut $ do
+            putWord8 firstByte
+            putByteString . getPublicKey . getKey $ paymentKey
+            putByteString . getPublicKey . getKey $ stakingKey
+      where
+          firstByte = case addressDiscrimination @Jormungandr discrimination of
+              RequiresNetworkTag -> 0x84
+              RequiresNoTag -> 0x04
+          expectedLength = 1 + 2*publicKeySize
+
+    deriveStakingPrivateKey (Jormungandr accXPrv) =
+        let
+            changeXPrv = -- lvl4 derivation; soft derivation of change chain
+                deriveXPrv DerivationScheme2 accXPrv (toEnum @(Index 'Soft _) 2)
+            stakeXPrv = -- lvl5 derivation; soft derivation of address index
+                deriveXPrv DerivationScheme2 changeXPrv (minBound @(Index 'Soft _))
+        in
+            Jormungandr stakeXPrv
+
+-- | 'NetworkDiscriminant' for Cardano TestNet & Jormungandr
+--
+-- @since 2.0.0
+jormungandrTestnet :: NetworkDiscriminant Jormungandr
+jormungandrTestnet = (RequiresNetworkTag, NetworkTag 0)
+
+{-------------------------------------------------------------------------------
+                                 Key generation
+-------------------------------------------------------------------------------}
+
+-- | Size, in bytes, of a public key (without chain code)
+publicKeySize :: Int
+publicKeySize = 32
+
+-- | Serialized length in bytes of a Single Address
+addrSingleSize :: Int
+addrSingleSize = 1 + publicKeySize
+
+-- | Serialized length in bytes of a Grouped Address
+addrGroupedSize :: Int
+addrGroupedSize = addrSingleSize + publicKeySize
+
+-- | Purpose is a constant set to 1852' (or 0x8000073c) following the BIP-44
+-- extension for Cardano:
+--
+-- https://github.com/input-output-hk/implementation-decisions/blob/e2d1bed5e617f0907bc5e12cf1c3f3302a4a7c42/text/1852-hd-chimeric.md
+--
+-- It indicates that the subtree of this node is used according to this
+-- specification.
+--
+-- Hardened derivation is used at this level.
+purposeIndex :: Word32
+purposeIndex = 0x8000073c
+
+-- | One master node (seed) can be used for unlimited number of independent
```

Comment:
Note that we don't want all these constants to show up in the documentation. Haddock will automatically prune these **if and only if** they have no docstring attached. So here, we need to replace `-- | ...` with `-- ...` for everything we don't want to be exposed.

Reply:
done